src/core/fee-calculation/recipient-recieved-scanner.ts

```
@@ -131,13 +131,22 @@ export class RecipientRecievedScanner {
 
     // Count only reward distributors for accurate progress tracking
     const rewardDistributors = distributorEntries.filter(
-      ([, info]) => info && info.is_reward_distributor,
+      ([, infoArray]) => infoArray && infoArray[0]?.is_reward_distributor,
     );
     let rewardDistributorIndex = 0;
 
     // Process distributors
-    for (const [address, distributorInfo] of distributorEntries) {
-      if (!distributorInfo) continue;
+    for (const [address, distributorInfoArray] of distributorEntries) {
+      if (!distributorInfoArray || distributorInfoArray.length === 0) continue;
+
+      const distributorInfo = distributorInfoArray[0];
+
+      if (!distributorInfo) {
+        console.log(
```

this should throw an error - since we already checked that the array is non empty it should only contain valid distributors

---

Fixed - Changed to throw an error instead of logging and continuing. The error now clearly indicates invalid distributor data.